lib/jsonapi/plugs/deserializer.ex

```diff
@@ -0,0 +1,66 @@
+defmodule JSONAPI.Deserializer do
+  @moduledoc """
+  This plug "deserializes" params for ease of use when casting to changesets.
```

Comment:
Assuming my above stated understanding is correct, it may be prudent to use the word "flatten" instead of "deserialize".